docs/source/frameworks/onnx.rst

```diff
@@ -0,0 +1,460 @@
+      def sr(img) -> np.ndarray:
+	  img = img.resize((224, 224))
+	  gray_img = ImageOps.grayscale(img)
+	  arr = np.array(gray_img) / 255.0 # convert from 0-255 range to 0.0-1.0 range
+	  arr = np.expand_dims(arr, (0, 1)) # add batch_size, color_channel dims
+	  sr_arr = runner.run.run(arr)
+	  sr_arr = np.squeeze(sr_arr) # remove batch_size, color_channel dims
+	  sr_img = PIL_Image.fromarray(np.uint8(sr_arr * 255) , 'L')
+	  return sr_img
+
+
+.. tab:: TensorFlow
+
+   .. code-block:: python
+
+      import bentoml
+
+      import numpy as np
+      from bentoml.io import Image
+      from bentoml.io import JSON
+
+      runner = bentoml.onnx.get("onnx_resnet50:latest").to_runner()
+
+      svc = bentoml.Service("onnx_resnet50", runners=[runner])
+
+      @svc.api(input=Image(), output=JSON())
+      def predict(img):
+
+	  from tensorflow.keras.applications.resnet50 import preprocess_input, decode_predictions
+
+	  img = img.resize((224, 224))
+	  arr = np.array(img)
+	  arr = np.expand_dims(arr, axis=0)
+	  arr = preprocess_input(arr)
+	  preds = runner.run.run(arr)
+	  return decode_predictions(preds, top=1)[0]
+
+
+.. tab:: Scikit Learn
+
+   TODO
+
+.. note::
+
+   In above codes we use ``runner.run.run(input_data)`` to do
+   inference. The first ``run`` is referring to
+   ``onnxruntime.InferenceSession``'s ``run`` method, while the second
+   ``run`` is BentoML's naming convention for doing runner inference
+   for a model method. For example, for a Keras model with ``predict``
+   method, we will call ``runner.predict.run(input_data)``.
+
+
+When constructing a :ref:`bentofile.yaml <concepts/bento:Bento Build Options>`,
+there are two ways to include ONNX as a dependency, via ``python`` or
+``conda``:
+
+.. tab-set::
+
+   .. tab-item:: python
+
+      .. code-block:: yaml
+
+         python:
+         - onnx
+	 - onnxruntime
+
+   .. tab-item:: conda
+
+      .. code-block:: yaml
+
+         conda:
+           channels:
+           - conda-forge
+           dependencies:
+           - onnx
+	   - onnxruntime
+
+
+Using Runners
+-------------
+
+.. seealso::
+
+   :ref:`Runners<concepts/runner:Using Runners>` for more information on what is
+   a Runner and how to use it.
+
+To use ``onnx`` runner locally, access the model via ``get`` and
+convert it to a runner:
+
+.. code-block:: python
+
+   test_input = np.random.randn(2, 1, 244, 244)
+
+   runner = bentoml.onnx.get("super_resolution").to_runner()
+
+   runner.init_local()
+
+   runner.run.run(test_input)
+
+.. note::
+
+   You don't need to cast your input ndarray to ``np.float32`` for
+   runner input
+
+Like ``load_model``, you can customize ``providers`` and
+``session_options`` when you create a runner:
+
+.. code-block:: python
+
+   providers=["TensorrtExecutionProvider", "CUDAExecutionProvider", "CPUExecutionProvider"]
+
+   runner = bentoml.onnx.get("onnx_super_resolution").with_options(providers=providers).to_runner()
+
+   runner.init_local()
```

Comment:
imo we don't need this part as we should always recommend user to use runner in a service context. A small section sayinig using `load_model` to test out the model inference similar to fastai should be good enough.